src/bindings/xfconf/elektra-xfconf-binding.c

```diff
@@ -0,0 +1,151 @@
+#include "elektra-xfconf-binding.h"
+#include "elektra-xfconf-channel.h"
+#include "elektra-xfconf-util.h"
+#include "elektra-xfconf.h"
+
+#define require_binding_read_lock() require_read_lock (&binding_lock, "BINDING")
+#define require_binding_write_lock() require_read_lock (&binding_lock, "BINDING")
+#define release_binding_lock() require_read_lock (&binding_lock, "BINDING")
+
+typedef struct
+{
+	gulong id;
+	XfconfChannel * channel;
+	const gchar * xfconf_property;
+	gpointer object;
+	const gchar * object_property;
+} propertyBinding;
+
+GList * property_bindings = NULL;
+gulong last_id = 0;
+
+pthread_rwlock_t binding_lock = PTHREAD_RWLOCK_INITIALIZER;
+
+static void notify_binding (const propertyBinding * binding, const GValue * gValue)
+{
+	trace ();
+	g_object_set_property (G_OBJECT (binding->object), binding->object_property, gValue);
+}
+
+gulong xfconf_g_property_bind (XfconfChannel * channel, const gchar * xfconf_property, GType xfconf_property_type, gpointer object,
+			       const gchar * object_property)
+{
+	trace ();
+	g_debug ("try to bind property %sto %s", xfconf_property, object_property);
+
+	propertyBinding * binding = calloc (1, sizeof (propertyBinding));
+	binding->channel = channel;
+	binding->xfconf_property = strdup (xfconf_property);
+	binding->object = object;
+	binding->object_property = strdup (object_property);
+
+	require_binding_write_lock () binding->id = last_id++;
```

Comment:
For nicer code formatting:

```suggestion
	require_binding_write_lock ();
	binding->id = last_id++;
```

Reply:
Should be done by 6506d0e23efa25047381ef0cd2afa5aed13da3f7 or before